taskcluster/app_services_taskgraph/transforms/script_to_run_task.py

```diff
@@ -0,0 +1,30 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# These transforms perform additional setup that could not have been done in the Dockerfile
+# due to this Taskcluster bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1587611
+
+from __future__ import absolute_import, print_function, unicode_literals
+from taskgraph.transforms.base import TransformSequence
+
+transforms = TransformSequence()
+
+
+def script_to_bash_command(script):
+    return [
+        "/bin/bash",
+        "--login",
+        "-c",
+        "cat <<'SCRIPT' > ../script.sh && bash -e ../script.sh\n{}\nSCRIPT".format(script)
```

Comment:
Blocker: I remember Mitch talked to me about this hack. We've made good progress since this code was first written. We can re-use the `run-commands` logic that was implemented in mobile projects.

Here's the logic: https://github.com/mozilla-mobile/fenix/blob/21606bf94f2cd2876841f904acf69ee450422c98/taskcluster/fenix_taskgraph/job.py#L51-L67
Here's how to use it: https://github.com/mozilla-mobile/fenix/blob/d719823b2f0a091224b9066006091469bd531754/taskcluster/ci/ui-test/kind.yml#L28-L31

I don't feel super confident in using this here-document. I foresee some shell quoting problems.

Reply:
Noted, will do.

Reply:
Chatted over Slack. Okay to tackle this in a follow up, due to the complexity of the task.

Reply:
Addressed in https://github.com/mozilla/application-services/pull/3113